src/dialogs/saveimagedialog.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2021 Meltytech, LLC
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "saveimagedialog.h"
+
+#include "mltcontroller.h"
+#include "settings.h"
+#include "util.h"
+
+#include <QDebug>
+#include <QtMath>
+
+static QString suffixFromFilter(const QString& filterText)
+{
+    QString suffix = filterText.section("*", 1, 1).section(")", 0, 0).section(" ", 0, 0);
+    if ( !suffix.startsWith(".") )
+    {
+        suffix.clear();
+    }
+    return suffix;
+}
+
+SaveImageDialog::SaveImageDialog(QWidget *parent, const QString& caption, QImage& image)
+    : QFileDialog(parent, caption)
+    , m_image(image)
+{
+    setModal(true);
+    setAcceptMode(QFileDialog::AcceptSave);
+    setFileMode(QFileDialog::AnyFile);
+    setOptions(Util::getFileDialogOptions());
+    QString path = Settings.savePath();
+    setDirectory(path);
+
+    QString nameFilter = tr("PNG (*.png);;BMP (*.bmp);;JPEG (*.jpg *.jpeg);;PPM (*.ppm);;TIFF (*.tif *.tiff);;WebP (*.webp);;All Files (*)");
+    setNameFilter(nameFilter);
+
+    QStringList nameFilters = nameFilter.split(";;");
+    QString suffix = Settings.snapSuffix();
+    QString selectedNameFilter = nameFilters[0];
+    for ( const auto& f : nameFilters )
+    {
+        if ( f.contains(suffix) )
+        {
+            selectedNameFilter = f;
+            break;
+        }
+    }
+    selectNameFilter(selectedNameFilter);
+
+    // Use the current player time as a suggested file name
+    QString nameSuggestion = tr("shotcut_%1").arg(MLT.producer()->frame_time( mlt_time_clock ));
```

Comment:
"shotcut" should be capitalized and not translatable.

Reply:
Done

I think I took some inspiration from VLC which uses the name: "vlcsnap-<timecode>.png". I debated adding a word like "export" or "frame". I do not have a strong opinion about this.  But perhaps some users will since it is not configurable and I would prefer to not add a user configuration for the name format.